cloud/amazon/ec2_asg_facts.py

```diff
@@ -299,7 +299,7 @@ def find_asgs(conn, module, name=None, tags=None):
         module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
 
     matched_asgs = []
-    name_prog = re.compile(r'^' + name)
+    name_prog = re.compile(r'^' + str(name))
```

Comment:
This would, if the name is `None`, make a regex `r'^None'` which could match groups people don't expect here. I'm going to fix this in a different patch with an explicit `None` check. Thanks for pointing out the error